### PR TITLE
fix: Remove deprecation warning on `Message#cleanContent`

### LINKED ISSUE #7142: Message.cleanContent sends a deprecation warning
### Issue description

The `Util.removeMentions` method is deprecated, but it is still used by `Util.cleanContent` which is used by [`Message.cleanContent`](https://github.com/discordjs/discord.js/blob/stable/src/structures/Message.js#L473)

So even though `Message.cleanContent` is not deprecated, it sends a deprecation warning on any call on a message that contains a mention.

Steps to reproduce:
1. Run the included code (event that logs the `cleanContent` of all recieved messages)
2. Send a message with a user mention

### Code sample

```typescript
const Discord = require('discord.js')
const client = new Discord.Client({ intents: ['GUILDS', 'GUILD_MESSAGES'] })

client
  .on('ready', () => {
    console.log(`Logged in as ${client.user.tag}!`)
  })
  .on('warn', console.log)
  .on('messageCreate', msg => {
    console.log(msg.cleanContent)
  })

client.login('token')
```

### discord.js version

13.4.0

### Node.js version

16.13.1

### Operating system

Windows

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

GUILDS, GUILD_MESSAGES

### I have tested this issue on a development release

_No response_

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
Resolves #7142.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Comment by Seblor** [comment]
I'm not familiar with the inner workings of DJS, but from an outsider's point of view, this looks like a stopgap solution.

The `Util.removeMentions` method is being deprecated because of the introduction of `allowedMentions` (https://github.com/discordjs/discord.js/pull/6530). So maybe the code from `Util.removeMentions` should be moved to the `Message.cleanContent` getter, or the `removeMentions` should not be deprecated.

**Comment by Jiralite** [comment]
`Message#cleanContent` calls `Util.cleanContent()` which calls `Util.removeMentions()` which sanitises input to prevent mentionability by a simple `str.replaceAll('@', '@\u200b')`. Once `Util.removeMentions()` is removed, this replacement will no longer exist. `Message#cleanContent` will still call `Util.cleanContent()` and will still act as it always has - the only difference being that mentions are not sanitised.

The reason for this is in case people send clean content as a new message which may cause `@everyone` or `@here` injections. Since `allowedMentions` exist, that should be the preferred method to prevent mentions in your message.

**Comment by Seblor** [comment]
That makes sense, thank you for the clarification.

**Comment by ImRodry** [comment]
Shouldn't cleanContent still return @Rodry#4020 instead of <@240875059953139714>? Maybe it shouldn't only remove mentions but also make them readable outside of Discord

**Comment by vladfrangu** [comment]
The only thing this PR does is prevent the deprecation warning from showing in .cleanContent, not remove the replace to username behavior

**Comment by ImRodry** [comment]
> Once `Util.removeMentions()` is removed, this replacement will no longer exist. `Message#cleanContent` will still call `Util.cleanContent()` and will still act as it always has - the only difference being that mentions are not sanitised.

I was commenting on this

**Comment by vladfrangu** [comment]
It means that if you cleanContent a member mention, whose display name is `everyone`, you will get an `@everyone` in your chat instead of `@\u200beveryone`.

File: src/util/Util.js [hunk 1]
```diff
@@ -529,7 +529,11 @@ class Util extends null {
    * @deprecated Use {@link BaseMessageOptions#allowedMentions} instead.
    */
   static removeMentions(str) {
-    if (!deprecationEmittedForRemoveMentions) {
+    return Util._removeMentions(str, false);
+  }
+
+  static _removeMentions(str, fromCleanContent) {
+    if (!deprecationEmittedForRemoveMentions && !fromCleanContent) {
       process.emitWarning(
```
Comment (kyranet) [inline src/util/Util.js:537]: This should be something like this:
```js
  static removeMentions(str) {
    if (!deprecationEmittedForRemoveMentions) {
      deprecationEmittedForRemoveMentions = true;
      process.emitWarning(
        // ...
      );
    }

    return Util._removeMentions(str);
  }

  static _removeMentions(str) {
    return str.replaceAll('@', '@\u200b');
  }
```

Likewise in `cleanContent`, remove the second parameter.

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/util/Util.js (hunk 1 above)
Comment (Jiralite) [reply src/util/Util.js:537]: > Also, does the lack of JSDoc automatically make this private/hidden from the docs?

Yep! By having no JSDoc, it's simply not parsed into the .json.

**Review by kyranet — APPROVED** [review]

**Pull request merged**